packages/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt

```diff
@@ -108,7 +139,9 @@ class ReactActivityDelegateWrapper(
   }
 
   override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
-    return delegate.onKeyUp(keyCode, event)
+    return reactActivityHandlers
+      .map { it.onKeyUp(keyCode, event) }
+      .fold(false) { accu, current -> accu || current } || delegate.onKeyUp(keyCode, event)
```

Comment:
it's tricky here and i always being confused if i did right. taking `onNewIntent` as an example:

```kotlin
  override fun onNewIntent(intent: Intent?): Boolean {
    val listenerResult = reactActivityLifecycleListeners
      .map { it.onNewIntent(intent) }
      .fold(false) { accu, current -> accu || current }
    val delegateResult = delegate.onNewIntent(intent)
    return listenerResult || delegateResult
  }
```

where i intentionally separate as two difference variables.  even if a listener returns true, i think we should still call `delegate.onNewIntent`.

in your case, if a listener returns true from onKeyUp, `ReactDelegate.onKeyUp` will not be called. i'm afraid there will be something broken from react-native's key handler.

Reply:
thanks for pointing this out, @Kudo -- I had initially written it this way on purpose because I thought the dev menu was meant to consume the key event and *not* pass it along to ReactDelegate. (That is how DevMenuAwareReactActivity is written.)

However, I tested out your version and it seems to make no difference at all to any dev menu related key events. So I will switch to this in order to match all the other similar listeners.

Reply:
(ignore my previous comment here, I messed up when I was testing 😅 )

Consuming the key event and not letting it through to ReactDelegate.onKeyUp is actually purposeful here, as this is the way that we currently override RN's built in dev menu. Otherwise the shake gesture brings up both RN's and our dev menu for a split second. DevMenuAwareReactActivity is [already written like this](https://github.com/expo/expo/blob/f311bf751d40952b1c55fb451877d1314cb73df8/packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/react/DevMenuAwareReactActivity.kt#L29) so I don't think this should cause issues from the dev menu specifically. But I understand your concern for the more general case 😕 I will at least document this in a comment, do you think that's sufficient?